### ISSUE TITLE
Web client - framework instrumentation

### ISSUE THREAD

**Issue by jpkrohling** [opentracing/opentracing-javascript#111]
This is a meta issue, probably as a follow up from #110.

To provide better support for web client tracing, it would be nice to understand what are the current stacks/frameworks being used and how we could transparently instrument them. Are there hooks we can add listeners/filters/interceptors?

The first part of this task is to identify the frameworks/stacks eligible for instrumenting. For each framework, open a new issue and link to this one.

**Comment by yurishkuro** [comment]
I would start by looking at zipkin-js, which already includes instrumentation for "popular" frameworks.

I am not sure about the emphasis on the "Web client" though - are you talking about tracing in the browser?

**Comment by jpkrohling** [comment]
> I am not sure about the emphasis on the "Web client" though - are you talking about tracing in the browser?

Yes, which might not be a browser. It could be a "web view" on mobile clients, for instance.

**Comment by yurishkuro** [comment]
I wonder if there are any open source tracers that work in the browser (since Jaeger doesn't).

**Comment by karunagoyalk** [comment]
@jpkrohling okay,I will work for #110 Thank you!

**Comment by jpkrohling** [comment]
@karunagoyalk , if you have progress to report on this issue, please do so before the deadline, which is tomorrow Tuesday, October 30, 2018 4pm UTC.

https://www.outreachy.org/december-2018-march-2019-outreachy-internships/communities/cncf-tracing/research-the-gaps-in-tracing-web-clients/contributions/
